src/tools/miri/src/shims/intrinsics/mod.rs

```diff
@@ -68,8 +68,8 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 throw_unsup_format!("unimplemented intrinsic: `{intrinsic_name}`")
             }
             let intrinsic_fallback_checks_ub = Symbol::intern("intrinsic_fallback_checks_ub");
-            if !this.tcx.item_attrs(instance.def_id()).iter().any(|attr| attr.path_matches(&[sym::miri, intrinsic_fallback_checks_ub])) {
-                throw_unsup_format!("miri can only use intrinsics that preserve UB. After verifying that `{intrinsic_name}` does so, add the `#[miri::intrinsic_fallback_checks_ub]` attribute to it");
+            if this.tcx.get_attrs_by_path(instance.def_id(), &[sym::miri, intrinsic_fallback_checks_ub]).next().is_none() {
+                throw_unsup_format!("miri can only use intrinsic fallback bodies that preserve UB. After verifying that `{intrinsic_name}` does so, add the `#[miri::intrinsic_fallback_checks_ub]` attribute to it");
```

Comment:
The previous code only worked for intrinsics not defined in the current crate, which was true for all intrinsics as they are defined in libcore

Reply:
`tcx.item_attrs` even has a comment saying "Do not use this directly, use tcx.get_attrs instead." But that's very easy to miss, maybe it should have a more scary name... or ideally it'd be private somehow.